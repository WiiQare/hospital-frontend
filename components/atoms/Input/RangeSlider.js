import { useState, useEffect, useRef } from "react";

const RangeSlider = ({ initialMin, initialMax, min, max, step, priceCap }) => {
  const progressRef = useRef(null);
  const [minValue, setMinValue] = useState(initialMin);
  const [maxValue, setMaxValue] = useState(initialMax);

  const handleMin = (e) => {
    if (maxValue - minValue >= priceCap && maxValue <= max) {
      if (parseInt(e.target.value) > parseInt(maxValue)) {
      } else {
        setMinValue(parseInt(e.target.value));
      }
    } else {
      if (parseInt(e.target.value) < minValue) {
        setMinValue(parseInt(e.target.value));
      }
    }
  };

  const handleMax = (e) => {
    if (maxValue - minValue >= priceCap && maxValue <= max) {
      if (parseInt(e.target.value) < parseInt(minValue)) {
      } else {
        setMaxValue(parseInt(e.target.value));
      }
    } else {
      if (parseInt(e.target.value) > maxValue) {
        setMaxValue(parseInt(e.target.value));
      }
    }
  };

  useEffect(() => {
    progressRef.current.style.left = (minValue / max) * step + "%";
    progressRef.current.style.right = step - (maxValue / max) * step + "%";
  }, [minValue, maxValue, max, step]);

  return (
    <div className="overflow-hidden">
      <div className="flex justify-between items-center my-6 ">
        <div className="rounded-md flex items-center">
          <span className="p-2 font-extralight text-xs"> Min</span>
          <label
            htmlFor=""
            className="text-sky font-bold text-2xl flex items-start"
          >
            {minValue}{" "}
            <span className="text-gray-500 font-light text-xs">$/mo</span>
          </label>
        </div>
        <div className="ml-2 font-semibold text-lg"> - </div>
        <div className=" ">
          <div className="rounded-md flex items-center">
            <span className="p-2 font-extralight text-xs"> Max</span>
            <label
              htmlFor=""
              className="text-sky font-bold text-2xl flex items-start"
            >
              {maxValue}
              <span className="text-gray-500 font-light text-xs">$/mo</span>
            </label>
          </div>
        </div>
      </div>

      <div className="mb-4">
        <div className="slider relative h-1 rounded-md bg-gray-300">
          <div
            className="progress absolute h-1 bg-green-300 rounded "
            ref={progressRef}
          ></div>
        </div>

        <div className="range-input relative  ">
          <input
            onChange={handleMin}
            type="range"
            min={min}
            step={step}
            max={max}
            value={minValue}
            className="range-min absolute w-full  -top-1  h-1   bg-transparent  appearance-none pointer-events-none"
          />

          <input
            onChange={handleMax}
            type="range"
            min={min}
            step={step}
            max={max}
            value={maxValue}
            className="range-max absolute w-full  -top-1 h-1  bg-transparent appearance-none  pointer-events-none"
          />
        </div>
      </div>
    </div>
  );
};

export default RangeSlider;
