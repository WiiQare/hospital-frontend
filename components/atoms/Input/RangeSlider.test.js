import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import RangeSlider from "./RangeSlider";

describe("RangeSlider", () => {
  let component;
  beforeEach(() => {
    const res = render(
      <RangeSlider
        initialMax={100}
        initialMin={10}
        min={0}
        max={1000}
        step={1}
        priceCap={100}
      />
    );
    component = res.container;
  });

  it("should render the component", () => {
    expect(component).toMatchSnapshot();
  });
});
