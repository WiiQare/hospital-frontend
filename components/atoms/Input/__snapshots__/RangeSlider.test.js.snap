// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RangeSlider should render the component 1`] = `
<div>
  <div
    class="overflow-hidden"
  >
    <div
      class="flex justify-between items-center my-6 "
    >
      <div
        class="rounded-md flex items-center"
      >
        <span
          class="p-2 font-extralight text-xs"
        >
           Min
        </span>
        <label
          class="text-sky font-bold text-2xl flex items-start"
          for=""
        >
          10
           
          <span
            class="text-gray-500 font-light text-xs"
          >
            $/mo
          </span>
        </label>
      </div>
      <div
        class="ml-2 font-semibold text-lg"
      >
         - 
      </div>
      <div
        class=" "
      >
        <div
          class="rounded-md flex items-center"
        >
          <span
            class="p-2 font-extralight text-xs"
          >
             Max
          </span>
          <label
            class="text-sky font-bold text-2xl flex items-start"
            for=""
          >
            100
            <span
              class="text-gray-500 font-light text-xs"
            >
              $/mo
            </span>
          </label>
        </div>
      </div>
    </div>
    <div
      class="mb-4"
    >
      <div
        class="slider relative h-1 rounded-md bg-gray-300"
      >
        <div
          class="progress absolute h-1 bg-green-300 rounded "
          style="left: 0.01%; right: 0.9%;"
        />
      </div>
      <div
        class="range-input relative  "
      >
        <input
          class="range-min absolute w-full  -top-1  h-1   bg-transparent  appearance-none pointer-events-none"
          max="1000"
          min="0"
          step="1"
          type="range"
          value="10"
        />
        <input
          class="range-max absolute w-full  -top-1 h-1  bg-transparent appearance-none  pointer-events-none"
          max="1000"
          min="0"
          step="1"
          type="range"
          value="100"
        />
      </div>
    </div>
  </div>
</div>
`;
